pkg/monitor/cluster/clusterwideproxystatus.go

```diff
@@ -164,7 +171,24 @@ func (mon *Monitor) emitCWPStatus(ctx context.Context) error {
 		for _, gatewayDomain := range clusterdetails.Spec.GatewayDomains {
 			gatewayDomain = strings.ToLower(gatewayDomain)
 			if !noProxyMap[gatewayDomain] {
-				missing_no_proxy_list = append(missing_no_proxy_list, gatewayDomain)
+				parts := strings.Split(gatewayDomain, ".")
+				domainfound := false
+				// Loop until there are at least 3 parts
+				for len(parts) >= 2 {
```

Comment:
Shouldn't this be >= 3, to align with the above comment (at least 3 parts)?

Reply:
Yeah, initially I started with 3 and later changed to 2 for testing. I think 2 is fine, its still an open item to discuss and finalize.

Reply:
We should just make sure the comments and the loop count actually matches because that's confusing.

Reply:
@hlipsig : sure thing, will go with 3 itself for the time being. If there is a necessity to reduce it, will do so at that time.